packages/tpetra/core/src/Tpetra_Details_WrappedDualView.hpp

```diff
@@ -98,28 +98,55 @@ using enableIfConstData = std::enable_if_t<hasConstData<DualViewType>::value>;
 template <typename DualViewType>
 using enableIfNonConstData = std::enable_if_t<!hasConstData<DualViewType>::value>;
 
+/* sync_host functions */
+  
 template <typename DualViewType>
 enableIfNonConstData<DualViewType>
 sync_host(DualViewType dualView) {
   // This will sync, but only if needed
   dualView.sync_host();
 }
 
+template <typename DualViewType>
+enableIfNonConstData<DualViewType>
+sync_host(const typename DualViewType::t_host::execution_space& exec, DualViewType dualView) {
+  // This will sync, but only if needed
+  dualView.sync_host();
+}
+
 template <typename DualViewType>
 enableIfConstData<DualViewType>
 sync_host(DualViewType dualView) { }
 
+template <typename DualViewType>
+enableIfConstData<DualViewType>
+sync_host(const typename DualViewType::t_host::execution_space& exec, DualViewType dualView) { } 
```

Comment:
According to https://github.com/kokkos/kokkos/blob/2d7715239700f50169bc50a96a234b05c28c9a2e/containers/src/Kokkos_DualView.hpp#L721-L736,  it seems that the `Kokkos::DualView` throws when calling `sync_<host/device>`. Is it intended that you change this behavior by making it a no-op ?